src/Paket.Core/Common/Async.fs

```diff
@@ -17,34 +18,47 @@ module AsyncExtensions =
             let! b'' = b'
             return (a'',b'')
         }
-
-    static member Choice(tasks : Async<'T option> seq) = async {
+    static member map f a =
+        async { return f a }
+    static member tryFind (f : 'T -> bool) (tasks : Async<'T> seq) = async {
```

Comment:
This is an interesting API.
It is almost like `Async.Choice`, but will return all the started tasks and an index which one matched the given function.